Jstree 2015

will-moore opened the pull request:
This is @dpwrussell's big https://github.com/openmicroscopy/openmicroscopy/pull/3141/ jsTree PR rebased onto develop (again)!

This is currently "breaking" since the web-UI will need a lot of testing to check what's broken.

The OmeroWeb integrations tests test_tree.py and test_show.py are passing, but require @joshmoore's workarounds at https://github.com/openmicroscopy/openmicroscopy/pull/3910
I have had to do a lot of merge conflict on these tests, since the previous work was before the ```lib.ITest``` work. cc @ximenesuk @sbesson.

Next I will work on Robot tests to get them passing before we go to manual testing of web scenarios.

NB: Many PRs (since last year) have conflicted with changes here and may have been broken in the merging above so lots of retesting needed! cc @aleksandra-tarkowska 

Rebased work:

 * gh-4081 (cc3df4f98020d1dae48cde6c4917371bcc1943aa)
 * gh-4082 (a4876000c03f8e2414da5e552072bf5ae73d1816)
 * gh-4114 (delete robot improvement)
 * gh-4124 (f7dbb4519cb67f1a4dc8602c7bea037c5597fbf2; fix trailing slash, for chgrp/)

--rebased-from #4112

aleksandra-tarkowska commented:
@will-moore many Flake8 issues

jburel commented:
Creation of new project not added automatically. Required refresh: Regression

jburel commented:
Out of synch:
 * Select a dataset with 5 images
 * Move to another dataset with 6 images
The central panel shows the first dataset instead of the second one.
See screenshot

<img width="860" alt="outofsynch" src="https://cloud.githubusercontent.com/assets/1022396/9136141/27544cd0-3d0c-11e5-96ca-dd55de8e1a9d.png">

will-moore commented:
I think the sync problem is because of images in multiple datasets
Root always seems to be in "All Members" mode, which might explain new Projects not showing in tree.

dpwrussell commented:
When you select a different item in the tree, it will then determine its parent to see if it needs to update the centre panel. In the case of the same dataset being in two projects and going from selecting one to the other, the centre panel would not update, but just change the selection appropriately. This error will almost certainly be there somewhere.

will-moore commented:
@jburel That commit should fix the "All Members" displayed for root, and seems to fix display of newly created Projects for root too.

will-moore commented:
@jburel I think the "Dataset sync" issue is actually just the image with the long name not showing up in that Dataset: roiFig-test thumbnails. The second of those 2 Datasets splitViewFig-test actually has rectangular images in it (not shown in your screenshot). I'll check why that image isn't showing up - seems to have no pixels (fake image?)

jburel commented:
@will-moore: could be. I will try with "true" images

aleksandra-tarkowska commented:
This PR caused integration test failure https://ci.openmicroscopy.org/view/Failing/job/OMERO-DEV-breaking-integration-web/322/testReport/ Could you please update them accordingly?

jburel commented:
```
./components/tools/OmeroWeb/test/integration/test_chgrp.py:229:80: E501 line too long (84 > 79 characters)
```

jburel commented:
Travis is happy Thanks @will-moore

will-moore commented:
@sbesson I fixed the test_permissions.py test to not leave a Tag in the public group: https://github.com/will-moore/openmicroscopy/commit/cca88e7c484f451ab2d74d7db0963494f397d76a
which seems to have fixed the web integrations tests:
https://ci.openmicroscopy.org/view/Failing/job/OMERO-DEV-breaking-integration-web/331/testReport/

jburel commented:
Another round of testing then we should be able to remove the breaking label. Thanks @will-moore

will-moore commented:
@jburel Is it time to remove the breaking flag for jstree now?
NB: Will also need to remove breaking flag on https://github.com/openmicroscopy/openmicroscopy/pull/3910 since we need that for projection query permissions.

jburel commented:
@will-moore: as mentioned in devteam, I wanted to double-check with @pwalczysko tomorrow since he might have to adjust some of the testing set up.
Then we can certainly remove the label.

will-moore commented:
@sbesson You'll also need to remove "breaking" label from https://github.com/openmicroscopy/openmicroscopy/pull/3910 too, since this PR needs that one.
 Thanks!

joshmoore commented:
Note: I've started a "Rebased work" section in the description.

aleksandra-tarkowska commented:
Robot failed https://ci.openmicroscopy.org/view/Failing/job/OMERO-DEV-merge-robotframework/76/robot/report/log.html

jburel commented:
more or less expected.

joshmoore commented:
Kicked due to travis/hdf5 failure.

joshmoore commented:
@jburel mentioned the plan to get this in earlier rather than later & follow-up with other PRs (big :thumbsup:), so I don't know if the points below are for this PR or one of those follow-ups but from a quick read:

 * Some of the methods are looking a bit long  (e.g. https://github.com/openmicroscopy/openmicroscopy/pull/3995/files?diff=unified&w=1#diff-9704a8389cb2e3a7dd3f753858c9a1baR407). It would be a good to eventually break them up and add unit tests.
 * The diff contains a few commented out lines that could be removed along with a few TODOs which I can't gauge.

aleksandra-tarkowska commented:
@will-moore cherry-pick https://github.com/aleksandra-tarkowska/openmicroscopy/commit/2857d6c91c2d2a329958495d9e125f45bbd2f735

test/integration/test_csrf.py .........................

======================================================= 25 passed in 44.03 seconds ========================================================

joshmoore commented:
@will-moore : travis warns that this won't be included tomorrow:

```
9.77s$ if [[ $BUILD == 'build-python' ]]; then ./components/tools/travis-build py-flake8; fi
./components/tools/OmeroWeb/test/integration/weblibrary.py:78:80: E501 line too long (89 > 79 characters)
./components/tools/OmeroWeb/test/integration/weblibrary.py:79:80: E501 line too long (84 > 79 characters)
./components/tools/OmeroWeb/test/integration/weblibrary.py:84:80: E501 line too long (94 > 79 characters)
./components/tools/OmeroWeb/test/integration/weblibrary.py:87:80: E501 line too long (86 > 79 characters)
./components/tools/OmeroWeb/test/integration/test_csrf.py:302:80: E501 line too long (94 > 79 characters)
./components/tools/OmeroWeb/test/integration/test_csrf.py:303:80: E501 line too long (99 > 79 characters)
```

aleksandra-tarkowska commented on components/tests/ui/testcases/web/rdef_test.txt:
```diff
@@ -6,7 +6,7 @@ Resource          ../../resources/web/login.txt
 
 Suite Setup         Run Keywords  User "${USERNAME}" logs in with password "${PASSWORD}"  Maximize Browser Window
-Suite Teardown      Close all browsers
+# Suite Teardown      Close all browsers
```
Could you remove hash?

will-moore replied:
Oops - Done, thanks

aleksandra-tarkowska commented on components/tools/OmeroWeb/omeroweb/decorators.py:
```diff
@@ -457,6 +457,9 @@ def wrapped(request, *args, **kwargs):
                     ctx.load_server_settings(conn, request)
 
                     share_id = kwargs.get('share_id')
+                    if share_id is None:
```
share_id should never been in request. what is exactly the usecase?

will-moore replied:
With the 'images' api, all IDs are passed in the query string. We just need to use the share_id in the decorator to get a share connection so that the share member can get a share connection to list the images in a share in the tree. 
https://github.com/will-moore/openmicroscopy/commit/dc06b68ca571c5142340a72ee0c94704a80b8f47

aleksandra-tarkowska replied:
I know what share_id is for. bu allowing ```share_id``` in query string you allow to establish share connection for every URL in webclient. That is a bug.

As discussed:
```
GET https://server/webclient/api/images/?share_id=...
```

should be replaced by 

```
https://server/webclient/api/share/share_id/
```

will-moore commented:
@aleksandra-tarkowska That reverts use of ?share_id in request to create share connection: https://github.com/will-moore/openmicroscopy/commit/0e64045ca740b972d748ab837e41bc0b8ee24538

pwalczysko commented:
This PR is excluded today because of error `` # PR 3995 will-moore 'Jstree 2015' (status: error)`` (due to Travis says @sbesson ) - would be good to have it in so that we can start wrapping up. @will-moore

will-moore commented:
@sbesson @pwalczysko @joshmoore @jburel I think this PR is pretty much ready to merge now. Anything else need mentioning / discussing / scheduling?

pwalczysko commented:
Confirmed the fixes from the testing sheet
https://docs.google.com/spreadsheets/d/10Go_WZNBm84vR9MqxQd6CHQhxmwq-B9XR5-Xh-fgCU4/edit#gid=2017453437
https://docs.google.com/spreadsheets/d/10Go_WZNBm84vR9MqxQd6CHQhxmwq-B9XR5-Xh-fgCU4/edit#gid=1075347298
and 
https://docs.google.com/spreadsheets/d/10Go_WZNBm84vR9MqxQd6CHQhxmwq-B9XR5-Xh-fgCU4/edit#gid=1391897821
and
https://docs.google.com/spreadsheets/d/10Go_WZNBm84vR9MqxQd6CHQhxmwq-B9XR5-Xh-fgCU4/edit#gid=15573525
the fixes from 
https://docs.google.com/spreadsheets/d/10Go_WZNBm84vR9MqxQd6CHQhxmwq-B9XR5-Xh-fgCU4/edit#gid=1731867412
were confirmed previously.

pwalczysko commented:
Will have a look at the trello card

pwalczysko commented:
There is still a desynchronization between tree and central pane in cases where the names of images end in a style such as [......#10]. Such images are sorted in the tree in a manner where #10 goes before #1, whereas in the central pane they are sorted "correctly". 

To repeat, navigate to http://eel.openmicroscopy.org/merge/webclient/?show=image-1861 (login as member-all-7, the "pagination" db must be deployed on eel) and use the down key to see the desynchronization.

will-moore commented:
@pwalczysko The sorting bug was purely upper-case vv lower-case issue, fixed by always sorting with lower-case. Also fixed the sorting in table, to handle items with same name (sort these by id).

pwalczysko commented:
OK. So after the new commits are in, the testing
1. Navigate to http://eel.openmicroscopy.org/merge/webclient/?show=dataset-201 , login member-all-7
2. Expand the dataset and check using the down arrow key click through every single image, on both pages, checking the sync
3. Switch to list view in central pane and check that the Name column shows just one arrow (because it s already sorted). 
4. Sort by Name, clicking on the sinlge arrow, and go through the whole dataset again (sort again on second page)
5. Cut all the images in the dataset without Paste (which will bring them into Orphaned) and repeat steps 2 - 4 above in Orphaned.

I think this test exhaustively covers all  the untested commits here.  Possibly we could manage a rebuild today ? @sbesson @joshmoore ?

pwalczysko commented:
2 more problems:
 1. the ``bruker`` images are still loaded out of sync between tree and central panel in case they are in a dataset (the bug goes away after sorting by Name, and is not present in Orphaned)
 2. Select an image in central pane, then click on another image (5 images highter) and then click onto the tree (on the selected image). After this, use the down key -> the next selected image (both in tree and central) will be the last image you clicked in the central pane on (= the one 5 images lower), not the next first lower image as expected

pwalczysko commented:
Another potential problem - after cut and paste of Bruker set inot Orphaned, when there already are some Brukers present, the newly pasted Burkers do not appear in the tree, just in the central pane (all sorts itself out after a refresh).

will-moore commented:
@pwalczysko The tree and thumbnail sorting should be fully sync'd now since I'm sorting the thumbnails on load. Also the thumbnail selection followed by hotkey should keep everything in sync too.

pwalczysko commented:
All good here, even the copy and paste of Bruker into the same folder twice without refresh is now fine https://github.com/openmicroscopy/openmicroscopy/pull/3995#issuecomment-139295502. Went trough orphaned and in-dataset workflow, multipage with bruker, also sorting by name and date, in orphaned and dataset. 
All works as expected.
As discussed with @will-moore the ``Unresponsive script...`` dialog problem when Cut and Paste whole Page of images will be sorted in another PR (ticket will follow during this morning when I investigate further).
Ready to merge.

joshmoore commented:
:tada: :balloon: :clap: :cake: :shipit: :rocket: :trophy: :beers:

will-moore commented:
Thanks all for testing and bug-fixes!

dpwrussell commented:
Thanks for all the hard work pushing this home, sorry I couldn't do it myself!

pwalczysko commented:
Next step : Does the webtagging work ? (looking)

pwalczysko commented:
1. The webtagging on eel merge does not work (after selection of Auto Tag, the central pane is just blank, impossible to even start)
2. The Tag Search seems to work fine, except some formatting issue in the tagging box
![screen shot 2015-09-11 at 13 32 29](https://cloud.githubusercontent.com/assets/2478303/9814914/e4e76d3e-5889-11e5-93c6-6c4cd652ad76.png)

dpwrussell commented:
Yeah, I knew it wouldn't but was waiting until it was merged to fix it.
Should only be a few minor tweaks required.
On Sep 11, 2015 8:35 AM, "pwalczysko" <notifications@github.com> wrote:

>
>    1. The webtagging on eel merge does not work (after selection of Auto
>    Tag, the central pane is just blank, impossible to even start)
>    2. The Tag Search seems to work fine, except some formatting issue in
>    the tagging box [image: screen shot 2015-09-11 at 13 32 29]
>    <https://cloud.githubusercontent.com/assets/2478303/9814914/e4e76d3e-5889-11e5-93c6-6c4cd652ad76.png>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openmicroscopy/openmicroscopy/pull/3995#issuecomment-139534416>
> .
>
